src/test/ui/lint/use_suggestion_json.rs

```diff
@@ -1,5 +1,5 @@
 // ignore-cloudabi
-// compile-flags: --error-format pretty-json -Zunstable-options
+// compile-flags: --error-format pretty-json -Zunstable-options --colorful-json=true
```

Comment:
I don't see a test for the _current_ output here :-/

Also, although I see the immediate benefit of this, I would like for us to tackle a more "general" solution at some point where we can also produce HTML, for example.

Reply:
We do have other `--error-format pretty-json` ui tests.

So you're imagining something like `--json-rendered=(plain|termcolor|html)`?

Reply:
ping @estebank

Reply:
In my mind something like `--json-rendered=extended` where the textual output would be the same and we'd have extra json fields annotating those with extra information:

```
{"text": "path `std::string::String`", "annotations": [{"start": 6, "end": 25, "highlight_type": "highlighted"}]}
```

That way any tool can consume these and turn them into whatever their needs may be.

That being said, having specific support for both terminal and html output make sense to me, given they probably cover 80% of all likely use cases.

Reply:
The `--json-rendered` list can be extended in the future to support schemes like the one you proposed. I will change the PR to only support `plain` and `termcolor` for now.